spec/cases/each_with_ar_sqlite.rb

```diff
@@ -29,11 +27,19 @@ class User < ActiveRecord::Base
 print "Parent: "
 puts User.first.name
 
+# https://www.sqlite.org/faq.html
+# Under Unix, you should not carry an open SQLite database across a fork() system call into the child process.
+ActiveRecord::Base.connection.disconnect!
+
 print "Parallel (#{in_worker_type}): "
 Parallel.each([1], in_worker_type => 1) do
+  ActiveRecord::Base.establish_connection(params)
```

Comment:
running with 0 disables threading/forking ... so there is no point in doing
the reconnect dance / it would be inefficient to do it, so we don't need to
worry about if something works on 0 or not.

On Fri, Mar 25, 2016 at 8:11 AM, Jacob Duffy <notifications@github.com>
wrote:

> In spec/cases/each_with_ar_sqlite.rb
> <https://github.com/grosser/parallel/pull/168#discussion_r57452692>:
>
> >  print "Parallel (#{in_worker_type}): "
> >  Parallel.each([1], in_worker_type => 1) do
> > +  ActiveRecord::Base.establish_connection(params)
>
> The reason I'm worried about breaking 0 with a given technique is that I
> can envision an application with optional Parallel use. If we demonstrate a
> technique to use Parallel with SQLite, I feel like it should work both ways.
>
> Personally I have my :test environment use 0 so my app which normally uses
> mysql can run it's specs with SQLite.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/grosser/parallel/pull/168/files/dd8484ce87d0b4dc43a5b61720d2266e5419059b#r57452692>
>